### PR TITLE
Filter API data by partner & remove SITE_ID from settings

### PR DESCRIPTION
**Pull request by vkaracic**
Reopening https://github.com/edx/course-discovery/pull/977 
This needs to be merged AFTER https://github.com/edx/ecommerce/pull/1368

FYI @edx/helio

### CODE REVIEW

**Review by afzaledx — DISMISSED** [review]
Just a reminder so that this does not get merged accidentally.

**Comment by afzaledx** [comment]
@vkaracic , can you please update this branch?

**Comment by afzaledx** [comment]
Update it again, please?

File: course_discovery/settings/base.py [hunk 1]
```diff
@@ -475,8 +476,6 @@
 
 DEFAULT_PARTNER_ID = None
 
-# See: https://docs.djangoproject.com/en/dev/ref/settings/#site-id
-SITE_ID = 1
```
Comment (tasawernawaz) [inline course_discovery/settings/base.py:479]: @vkaracic @afzaledx  Why we have removed this from settings? We have seen `222` errors (`django.contrib.sites.models:DoesNotExist: Site matching query does not exist.`)on stage between `11:16 AM` and `11:26 AM`.

For details see: https://rpm.newrelic.com/accounts/88178/applications/15120497/filterable_errors#/show/cfc6f32e-6ab8-11e7-8968-0242ac11000f_0_3554/stack_trace?top_facet=transactionUiName&primary_facet=error.class&barchart=barchart&_k=1kc53y

Comment (afzaledx) [reply course_discovery/settings/base.py:479]: @tasawernawaz, is this from today?

Comment (tasawernawaz) [reply course_discovery/settings/base.py:479]: @afzaledx Yes, Please see new relic link.

Comment (afzaledx) [reply course_discovery/settings/base.py:479]: The newrelic link shows 6th July, that's why I asked.

Comment (tasawernawaz) [reply course_discovery/settings/base.py:479]: @afzaledx Sorry that link was not correct. See updated link https://rpm.newrelic.com/accounts/88178/applications/15120497/filterable_errors#/show/cfc6f32e-6ab8-11e7-8968-0242ac11000f_0_3554/stack_trace?top_facet=transactionUiName&primary_facet=error.class&barchart=barchart&_k=1kc53y

Comment (waheedahmed) [reply course_discovery/settings/base.py:479]: @vkaracic @afzaledx There is another error spike due to this, seen 196 times on stage
https://rpm.newrelic.com/accounts/88178/applications/15120497/filterable_errors#/table?top_facet=transactionUiName&barchart=barchart&_k=wrsl9q

**Pull request merged**